cpp/src/arrow/io/buffered_test.cc

```diff
@@ -297,6 +297,19 @@ TEST_F(TestBufferedOutputStream, Tell) {
   AssertTell(0);
 }
 
+TEST_F(TestBufferedOutputStream, Tell2) {
+  OpenBuffered();
+  AssertTell(0);
+
+  const std::string datastr = GenerateRandomData(10000);
```

Comment:
A simple test string should be enough. It costs time to generate random data.
I think we can move this case to existing `Tell` tests, not necessary for a new test.